website/src/docs/reference/templating.md

```diff
@@ -0,0 +1,912 @@
+
+#### `USER_WORKING_DIR`
+- **Type**: `string`
+- **Description**: Absolute path where `task` was called from
+
+```yaml
+tasks:
+  info:
+    cmds:
+      - echo "Root: {{.ROOT_DIR}}"
+      - echo "Current: {{.TASKFILE_DIR}}"
+      - echo "Working: {{.USER_WORKING_DIR}}"
+```
+
+### Status Context
+
+#### `CHECKSUM`
+- **Type**: `string`
+- **Description**: Checksum of files in `sources` (only in `status` with `checksum` method)
+
+#### `TIMESTAMP`
+- **Type**: `time.Time`
+- **Description**: Greatest timestamp of files in `sources` (only in `status` with `timestamp` method)
+
+```yaml
+tasks:
+  build:
+    method: checksum
+    sources: ["**/*.go"]
+    status:
+      - test "{{.CHECKSUM}}" = "$(cat .last-checksum)"
+    cmds:
+      - go build ./...
+      - echo "{{.CHECKSUM}}" > .last-checksum
+```
+
+### Loop Context
+
+#### `ITEM`
+- **Type**: `any`
+- **Description**: Current iteration value when using `for` property
+
+```yaml
+tasks:
+  greet:
+    cmds:
+      - for: [alice, bob, charlie]
+        cmd: echo "Hello {{.ITEM}}"
+```
+
+Can be renamed using `as`:
+
+```yaml
+tasks:
+  greet:
+    cmds:
+      - for:
+          var: NAMES
+          as: NAME
+        cmd: echo "Hello {{.NAME}}"
+```
+
+### Defer Context
+
+#### `EXIT_CODE`
+- **Type**: `int`
+- **Description**: Failed command exit code (only in `defer`, only when non-zero)
+
+```yaml
+tasks:
+  deploy:
+    cmds:
+      - ./deploy.sh
+      - defer: |
+          {{if .EXIT_CODE}}
+          echo "Deployment failed with code {{.EXIT_CODE}}"
+          ./rollback.sh
+          {{end}}
+```
+
+### System Context
+
+#### `TASK_VERSION`
+- **Type**: `string`
+- **Description**: Current version of Task
+
+```yaml
+tasks:
+  version:
+    cmds:
+      - echo "Using Task {{.TASK_VERSION}}"
+```
+
+## Built-in Functions
+
+These functions are provided by Go's [text/template](https://pkg.go.dev/text/template#hdr-Functions) package.
+
+### Logic Functions
+
+#### `and`
+Boolean AND operation
+```yaml
+cmds:
+  - echo "{{if and .DEBUG .VERBOSE}}Debug mode{{end}}"
+```
+
+#### `or`
+Boolean OR operation
+```yaml
+cmds:
+  - echo "{{if or .DEV .STAGING}}Non-production{{end}}"
+```
+
+#### `not`
+Boolean negation
+```yaml
+cmds:
+  - echo "{{if not .PRODUCTION}}Development build{{end}}"
+```
+
+### Data Access
+
+#### `index`
+Access array/map elements
+```yaml
+vars:
+  SERVICES: [api, web, worker]
+cmds:
+  - echo "First service: {{index .SERVICES 0}}"
+```
+
+#### `len`
+Get length of arrays, maps, or strings
+```yaml
+vars:
+  ITEMS: [a, b, c, d]
+cmds:
+  - echo "Found {{len .ITEMS}} items"
+```
+
+#### `slice`
+Get slice of array/string
+```yaml
+vars:
+  ITEMS: [a, b, c, d, e]
+cmds:
+  - echo "{{slice .ITEMS 1 3}}" # [b c]
+```
+
+### Output Functions
+
+#### `print`, `printf`, `println`
+Formatted output functions
+```yaml
+cmds:
+  - echo "{{printf "Version: %s.%d" .VERSION .BUILD}}"
+```
+
+## Slim-Sprig Functions
```

Comment:
I actually don't think most of our users know (or care) what the difference between built-in, slim-sprig and task-specific functions are? I wonder if its worth just merging them all for simplicity.

It makes more sense to group them logically then by where they are defined in some cases. E.g. `randInt` and `randIntN` are in entirely different places despite being logically very similar because they're defined in different places.

Reply:
I agree ! I pushed new version !